src/main/java/io/openliberty/tools/common/plugins/config/ServerConfigDocument.java

```diff
@@ -121,6 +121,12 @@ public ServerConfigDocument(CommonLoggerI log, File serverXML, File configDir, F
         initializeAppsLocation(log, serverXML, configDir, bootstrapFile, bootstrapProp, serverEnvFile, giveConfigDirPrecedence, libertyDirPropertyFiles);
     }
 
+    public ServerConfigDocument(CommonLoggerI log) {
+        this.log = log;
+        props = new Properties();
+        defaultProps = new Properties();
```

Comment:
I think you also need to initialize `libertyDirectoryPropertyToFile`. It is used in `VariableUtility.getPropertyValue` by one of the methods you made public below.